src/dd_init.F90

```diff
@@ -0,0 +1,326 @@
+module dd_init
+
+  use constants
+  use dd_header,  only: dd_type
+  use error,      only: fatal_error, warning
+  use global,     only: n_procs, n_particles, rank, mpi_err
+  use mesh,       only: bin_to_mesh_indices, mesh_indices_to_bin
+  use output,     only: write_message
+  use search,     only: binary_search
+  use string,     only: to_str
+
+#ifdef MPI
+  use mpi
+#endif
+
+  implicit none
+  public
+
+contains
+
+!===============================================================================
+! INITIALIZE_DOMAIN_DECOMP
+!===============================================================================
+
+  subroutine initialize_domain_decomp(dd)
+
+    type(dd_type), intent(inout) :: dd
+
+    integer :: d                ! neighbor bin
+    integer :: neighbor_meshbin
+    integer :: alloc_err        ! allocation error code
+    integer :: world_group
+    integer :: domain_master_group
+
+    call write_message("Initializing domain decomposition parameters...", 6)
+
+#ifdef _OPENMP
+    call fatal_error("Domain decomposition not implemented in " // &
+         "conjunction with OpenMP.")
```

Comment:
Thank you, @smharper . I found Travis tests failed because the two domain decomposition tests give errors when the code is compiled with openMP. So what's your suggestion to exclude these two tests for omp-compiling modes? Thanks.